File: LinkedList/ReverseLL.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

""" Iterative Solution """
class Solution:
    # @param {ListNode} head
    # @return {ListNode}
    def reverseList(self, head):
        current = head
        previous = None
        temp = head
        
        while current: # while current != None:
            temp = current.next
            current.next = previous
            
            previous = current
            current = temp
        return previous

File: LinkedList/ReverseLL.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode current = head;
        ListNode previous = null;
        ListNode temp = current;
        
        while (current != null) {
            temp = current.next;
            current.next = previous;
            
            previous = current;
            current = temp;
        }
        return previous;
    }
}
